docs/readmes/bazel/agw_with_bazel.md

```diff
@@ -0,0 +1,523 @@
+- `bazel/`
+    - Centralized folder for custom Bazel definitions and configuration of external sources.
+- `bazel/scripts/`
+    - Centralized folder for Bash wrapper scripts for Bazel, e.g. for running the [Starlark formatter](#starlark-formatter), [Bazel-diff](#bazel-diff) or the [LTE integration tests](#lte-integration-tests).
+- `bazel/bazelrcs/name.bazelrc`
+    - Configuration of flags for commands in different environments.
+- `bazel/name.bzl`
+    - Custom Bazel code - these are custom [Bazel rules](https://bazel.build/extending/rules), configuration of the tool chain and external dependencies.
+    - In `cpp`, `go` and `python_repositories.bzl` are external repositories specified. These can be for example local repositories (folder in the environment) or git repositories.
+- `bazel/external/name.BUILD`
+    - BUILD files for external repositories that are not bazelified.
+
+#### Starlark formatter
+
+To ensure the standardized formatting of all `BUILD.bazel` and `.bzl` files the Starlark formatter [buildifier](https://github.com/bazelbuild/buildtools/tree/master/buildifier) is used and run in CI.
+
+You can also run the Starlark format check locally, from your host or from any [Magma environment](#environments). To run the buildifier from your host system you need to export `MAGMA_ROOT` or navigate to the root of the repository first.
+
+- To run the Starlark format check locally, run `$MAGMA_ROOT/bazel/scripts/run_buildifier.sh check`
+- To run the Starlark formatter and to fix most errors that occur during the check, run `$MAGMA_ROOT/bazel/scripts/run_buildifier.sh format`
+
+### Environments
+
+The following Magma environments currently support Bazel:
+
+- Docker
+    - **devcontainer**
+        - The devcontainer is defined in `.devcontainer/Dockerfile` and meant for local development. The docker image is published at [`ghcr.io/magma/magma/devcontainer:latest`](https://github.com/magma/magma/pkgs/container/magma%2Fdevcontainer).
+    - **Bazel-base**
+        - The Bazel-base docker container is a minimal container where only required dependencies are installed. While this image is mainly intended to be used in CI, it can also be useful for local testing. The Bazel-base image is defined in `.devcontainer/bazel-base/Dockerfile` and published at [`ghcr.io/magma/magma/bazel-base:latest`](https://github.com/magma/magma/pkgs/container/magma%2Fbazel-base).
+- Vagrant VMs
+    - **magma-dev**
+        - The magma-dev virtual machine is intended for local development and for extended testing in CI. It can be used to run the Magma AGW services, as well as e.g. the [Python sudo tests](#python-sudo-tests).
+    - **magma-deb**
+        - The magma-deb virtual machine comes with an installed Magma debian package. This VM is used in CI to run the [LTE integration tests](#lte-integration-tests) against the AGW services, but it can also be used for local development.
```

Comment:
Imho this vm can not be used for local development - a lot of tools are missing - including Bazel. This machine is more designed to provide a minimalistic environment for installing a magma debian artifact (+ making it ready for integration testing).

Reply:
Oh, good to know - should we remove it entirely or just clarify this point? There are several other places in this file where this needs to be adapted.

Reply:
I would remove it.

Reply:
Pushed